### ISSUE TITLE
[Issue] Generate all .d.ts files in place instead of generate them in dist

### ISSUE THREAD

**Issue by gtrabanco** [stacksjs/bun-plugin-dts-auto#5]
### Describe the bug

After applying https://github.com/stacksjs/bun-plugin-dts-auto/issues/4#issuecomment-1666842824 This patch to solve the issues with the package all `.d.ts` files are created in the path where the files are.

![Captura de Pantalla 2023-08-06 a las 14 31 02](https://github.com/stacksjs/bun-plugin-dts-auto/assets/1969593/560c52d4-d598-4a21-8e63-ce45ae313d7a)

Instead of creating all those `.d.ts` files in the `outDir`:

![Captura de Pantalla 2023-08-06 a las 14 32 06](https://github.com/stacksjs/bun-plugin-dts-auto/assets/1969593/3de6ceaf-bf08-4abe-846a-fe0f9b250fcd)

Not sure if this is the expected behaviour.

### Reproduction

```ts
#!/usr/bin/env bun --bun
import dts from 'bun-plugin-dts-auto';

await Bun.build({
  entrypoints: [
    './src/index.ts',
  ],
  target: 'node',
  sourcemap: "external",
  minify: true,
  root: './src',
  outdir: './dist',
  plugins: [
    dts({
      withSourceMap: true, // optional
    }),
  ],
});

console.log('Build complete ✅');
```

### System Info

```Shell
System:
    OS: macOS 12.6.7
    CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
    Memory: 42.22 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 20.4.0 - ~/Library/Caches/fnm_multishells/45496_1690908490075/bin/node
    npm: 9.7.2 - ~/Library/Caches/fnm_multishells/45496_1690908490075/bin/npm
    pnpm: 8.6.6 - ~/.bun/bin/pnpm
  Browsers:
    Brave Browser: 115.1.56.14
    Safari: 16.5.2


Bun: 0.7.3 (today's latest canary).
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/stacksjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/stacksjs/contributing).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by mfeckie** [comment]
I fixed this by updating the `tsconfig.json` to tell it the correct output location for the .d.ts files

```
{
  "compilerOptions": {
   //
    "outDir": "./dist",
}
```

**Comment by chrisbbreuer** [comment]
Thanks for creating this issue! 

My apologies on the late resolution here, but we are now finally able to support this more actively now. Stacks had been consuming all of my free time, but this plugin is now finally a first-class citizen _(which means great ongoing support  😃)_

This is resolved in v0.4.0 by treating the `tsconfig.json` as a default which can be overriden by the `outDir` option.

Let me know if there is anything else I can help with!

Cheers ✌️


https://github.com/stacksjs/bun-plugin-dts-auto/compare/v0.3.0...v0.4.0
